cmd/evm/internal/t8ntool/transition.go

```diff
@@ -335,9 +342,23 @@ func signUnsignedTransactions(txs []*txWithKey, signer types.Signer) (types.Tran
 		v, r, s := tx.RawSignatureValues()
 		if key != nil && v.BitLen()+r.BitLen()+s.BitLen() == 0 {
 			// This transaction needs to be signed
-			signed, err := types.SignTx(tx, signer, key)
-			if err != nil {
-				return nil, NewError(ErrorJson, fmt.Errorf("tx %d: failed to sign tx: %v", i, err))
+			var (
+				signed *types.Transaction
+				err    error
+			)
+			if txWithKey.protected {
+				signed, err = types.SignTx(tx, signer, key)
+				if err != nil {
+					return nil, NewError(ErrorJson, fmt.Errorf("tx %d: failed to sign tx: %v", i, err))
+				}
+			} else {
+				unprotectedSigner := types.FrontierSigner{}
+				h := unprotectedSigner.Hash(tx)
+				sig, err := crypto.Sign(h[:], key)
```

Comment:
This is a bit odd way of doing it. How come you use this way of signing instead of `types.SignTx(tx, types.FrontierSigner{}, key)` ?

Reply:
You're right - very odd. For some reason, I had it in my head that the `SignTx` method only supported signing protected txs. Clearly I was wrong! Updated.